mne/decoding/transformer.py

```diff
@@ -245,6 +245,97 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(TransformerMixin):
+    """Transforms n-dimensional array into 2D array of n_sample by n_features.
+
+    This class reshapes an n-dimensional array into an n_sample * n_features
+    array, usable by the estimators and transformers of scikit-learn.
+
+    Examples
+    --------
+    clf = make_pipeline(SpatialFilter(), XdawnTransformer(), Vectorizer(),
+                        LogisticRegression())
+
+    Attributes
+    ----------
+    ``shape_`` : array
+         Stores the original dimension of data.
+    """
+
+    def fit(self, X, y=None):
+        """Stores the shape of X.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to fit. Can be, for example a list, or an array of at
+            least 2d. The first dimension must be of shape (n_samples).
```

Comment:
Although `n_samples` is used everywhere in sklearn, here - in mne-python it might be confused with timesamples. I think it is worth some clarification just in case.